### PR TITLE
Feature/task 125 - Id automatic generation for tasks

### PR DESCRIPTION
**Pull request by strymsg**
Finished the generation of task with an unique id. The id format follows:
 
```
task_{type}_{uuid4}_{creation_time in format ("%m/%d/%Y %H:%M:%S")}

For example

task_Api-request_63f1bd71-f441-4519-8a41-44643ccb4dad_04/27/2021 13:35:35
```

### CODE REVIEW

File: task_scheduler/tasks/abstract_task.py [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
         self.creation_time = datetime.now()
         self.type = type
         self.execution_information = None
-        # TODO: Define how to get the task_id
-        self._task_id = ''
+
+        # For instance:
+        # task_Api-request_63f1bd71-f441-4519-8a41-44643ccb4dad_04/27/2021 13:35:35
+        self._task_id = \
+            f'task_{type}_{uuid.uuid4()}_{self.creation_time.strftime("%m/%d/%Y %H:%M:%S")}'
```
Comment (cristiang96) [inline task_scheduler/tasks/abstract_task.py]: It's a good id generation, but for our application, I think We might only need the "uuid1().hex" representation, like this:
  
    # For instance:
    # task_Api-request_6b10a16e02e711e8ae5900163e990bdb
    self._task_id = f'task_{type}_{uuid.uuid1().hex}'

We would be avoiding having spaces in the "id" generation, and the id generation is made by using MAC address and time.

**Review by cristiang96 — COMMENTED** [review]
It's a good id generation, but for our application, I think We might only need the "uuid1().hex" representation, like this:
  
    # For instance:
    # task_Api-request_6b10a16e02e711e8ae5900163e990bdb
    self._task_id = f'task_{type}_{uuid.uuid1().hex}'

We would be avoiding having spaces in the "id" generation, and the id generation is made by using MAC address and time.

File: task_scheduler/tasks/abstract_task.py (hunk 1 above)
Comment (strymsg) [reply task_scheduler/tasks/abstract_task.py]: Changed to:
```python
# task_Api-request_63f1bd71-f441-4519-8a41-44643ccb4dad
self._task_id = f'task_{type}_{uuid.uuid4()}'
```

Comment (cristiang96) [reply task_scheduler/tasks/abstract_task.py]: Great, You could now merge to "develop" branch. @strymsg

**Pull request merged**